app/schema_validation/__init__.py

```diff
@@ -1,10 +1,21 @@
 import json
 
-from jsonschema import Draft4Validator, ValidationError
+from jsonschema import (Draft4Validator, ValidationError, FormatChecker)
+from notifications_utils.recipients import (validate_phone_number, validate_email_address)
 
 
 def validate(json_to_validate, schema):
-    validator = Draft4Validator(schema)
+    format_checker = FormatChecker()
+
+    @format_checker.checks('phone_number')
+    def validate_schema_phone_number(instance):
+        return validate_phone_number(instance)
+
+    @format_checker.checks('email_address')
+    def validate_schema_email_address(instance):
+        return validate_email_address(instance)
```

Comment:
i think these should catch `InvalidEmail` and `InvalidPhone`, and have this function just return a boolean of "is the email address valid" - if you provide a bad template id and no phone number, then it returns you a `ValidationError` that crucially contains both of those messages - if you provide a bad template id and a _bad_ phone number it returns an `InvalidPhoneError` and does not mention your bad template id. that way `validate` will only ever throw `jsonschema.ValidationError` which is nicer, I think.